packages/@ember/-internals/routing/lib/services/routing.ts

```diff
@@ -54,21 +54,25 @@ export default class RoutingService extends Service {
 
   generateURL(routeName: string, models: {}[], queryParams: {}) {
     let router = this.router;
-    // Return early when transition has not started, when rendering in tests without visit(),
-    // we cannot infer the route context which <LinkTo/> needs be aware of
-    if (!router._initialTransitionStarted) {
-      return;
+    try {
+      let visibleQueryParams = {};
+      if (queryParams) {
+        assign(visibleQueryParams, queryParams);
+        this.normalizeQueryParams(routeName, models, visibleQueryParams as QueryParam);
+      }
+
+      return router.generate(routeName, ...models, {
+        queryParams: visibleQueryParams,
+      });
+    } catch (e) {
+      // Swallow error when transition has not started.
+      // When rendering in tests without visit(), we cannot infer the route context which <LinkTo/> needs be aware of
+      if (!router._initialTransitionStarted) {
+        return;
+      } else {
+        throw e;
+      }
```

Comment:
Let's avoid the try/catch when we have started a transition, if we don't do that we break the ability to use the "break on uncaught exception" in the devtools.

I think a plausible solution here is something like:

```js
_generateURL(routeName: string, models: {}[], queryParams: {}) {

}

generateURL(routeName: string, models: {}[], queryParams: {}) {
  let router = this.router;

  if (router._initialTransitionStarted) {
    try {
      return this._generateURL(routeName, models, queryParams);
    } catch (error) {
      return;
    }
  } else {
    return this._generateURL(routeName, models, queryParams);
  }
}
```